### PR TITLE
Break _createDialog, _createInstance, _createForm into async steps

### PR DESCRIPTION
**Pull request by megheaiulian**
Break _createDialog, _createInstance,  _createForm into small async steps

### CODE REVIEW

**Review by nomego — APPROVED** [review]

**Comment by megheaiulian** [comment]
Tests are hanging because of Polymer/web-component-tester#648. Will downgrade in `package.json` to `6.3.0`.

**Review by plequang — CHANGES_REQUESTED** [review]
In Polymer 1.x, all properties forwared in `_forwardParentProp` were forwared after the element was attached, so after the dialog template instance was created, and we wrere able to forward the properties to the template instance.
This is not the case anymore, so we should do the same as for Polymer 2.x, by storing these properties in `_pendingProps` and forward these after template instance creation

File: cosmoz-dialog-behavior.html [hunk 1]
```diff
@@ -42,29 +43,37 @@
...
 
 		created: function () {
-			this._instanceProps = {};
-			this._parentModel = true;
 			this._dialogEventsHandler = this._handlePaperDialogEvents.bind(this);
-			this._pendingProps = [];
+			this._pendingProps = {};
+			this._spawnSteps = [this._createDialog, this._createInstance];
+			this._spawn = this._spawn.bind(this);
+			this._ensureTemplatized();
```
Comment (plequang) [inline cosmoz-dialog-behavior.html]: Don't know why, but `_ensureTemplatized` cannot be called during `created`.
When `templatize` is called during created, it seems Polymer does not find any properties to forward.
It works when `templatize` is called during `attached` though.

File: cosmoz-dialog-behavior.html [hunk 2]
```diff
@@ -150,32 +160,61 @@
...
 
+		_createInstance: function () {
+			if (!this._paperDialog) {
+				return;
+			}
+			const instance = this.stamp(),
+				props = this._pendingProps;
+
+			Object.keys(props).forEach((key)=>{
+				instance.forwardHostProp(key,props[key]);
```
Comment (plequang) [inline cosmoz-dialog-behavior.html]: In Polymer 1.x, this should be `instance[key] = ...`

**Review by plequang — CHANGES_REQUESTED** [review]
See review comments.
There are also some eslint errors.

File: cosmoz-dialog-behavior.html (hunk 2 above)
Comment (megheaiulian) [reply cosmoz-dialog-behavior.html]: Yes. `forwardHostProp` not available in 1.x

File: cosmoz-dialog-behavior.html (hunk 1 above)
Comment (megheaiulian) [reply cosmoz-dialog-behavior.html]: @plequang To do this correctly we need first create a mutation observer and in it set `_userTemplate` to the first found template then continue to templatize and _createInstance and _createDialog.

Comment (plequang) [reply cosmoz-dialog-behavior.html]: OK

**Review by plequang — APPROVED** [review]

**Pull request merged**